### DISCUSSION TITLE
2.40.0-rc.0 / 2022-11-03

### DISCUSSION THREAD

**Discussion by codesome** [prometheus/prometheus#11531] [Announcements]
This release introduces an experimental, native way of representing and storing histograms.

It can be enabled in Prometheus via `--enable-feature=native-histograms` to accept native histograms. Refer the documentation for more information on how to query native histograms. The docs for this are present close to the relevant sections, for example querying, API, etc. 
Enabling native histograms will also switch the preferred exposition format to protobuf.

To instrument your application with native histograms, use the `main` branch of `client_golang` (this will change for the final release when v1.14.0 of client_golang will be out), and set the `NativeHistogramBucketFactor` in your `HistogramOpts` (`1.1` is a good starting point).
Your existing histograms won't switch to native histograms until `NativeHistogramBucketFactor` is set.

* [FEATURE] Add **experimental** support for native histograms. Enable with the flag `--enable-feature=native-histograms`. #11447
* [FEATURE] SD: Add service discovery for OVHcloud. #10802
* [ENHANCEMENT] Kubernetes SD: Use protobuf encoding. #11353
* [ENHANCEMENT] TSDB: Use golang.org/x/exp/slices for improved sorting speed. #11054 #11318 #11380
* [ENHANCEMENT] Consul SD: Add enterprise admin partitions. Adds `__meta_consul_partition` label. Adds `partition` config in `consul_sd_config`. #11482
* [BUGFIX] API: Fix API error codes for `/api/v1/labels` and `/api/v1/series`. #11356 

<hr /><em>This discussion was created from the release <a href='https://github.com/prometheus/prometheus/releases/tag/v2.40.0-rc.0'>2.40.0-rc.0 / 2022-11-03</a>.</em>

**Comment by ArthurSens** [comment]
> Refer the documentation for more information on how to query native histograms.

Would it make sense to add a link to the documentation in the release? 😅

**Reply by codesome** [reply]
It is actually spread at multiple places for their particular sections. For example, if you want to know how to query native histograms, you would go to the querying docs and search for native histograms there.

There are "Note" all around the docs mentioning how things change and if something is not supported for native histograms. For example, if you are wondering if recording rules that produce native histograms work, you can head over to the recording rules docs and it will tell you there.

**Reply by codesome** [reply]
Updated the release notes saying you will find the docs at relevant places. For the final release however, there will likely be a blog post with a lot of info at a single place to get started.

**Reply by ArthurSens** [reply]
Oh I see, thanks for the info ❤️!

**Comment by Br0kenx** [comment]
I need to make money but I don't know how
